src/ClientInstance.cpp

```diff
@@ -384,14 +385,27 @@ PVR_ERROR CVNSIClientInstance::GetChannels(bool radio, kodi::addon::PVRChannelsR
       if (m_protocol >= 6)
       {
         std::string path = CVNSISettings::Get().GetIconPath();
-        std::string ref = vresp->extract_String();
+        std::string ref = vresp->extract_String(); // Picon ref
         if (!path.empty())
         {
           if (path[path.length() - 1] != '/')
             path += '/';
-          path += ref;
-          path += ".png";
-          tag.SetIconPath(path);
+
+          ref += ".png";
+          std::string image_path = path + ref;
+          bool exists = kodi::vfs::FileExists(image_path);
+          if (exists) // Use picon
+          {
+            tag.SetIconPath(image_path);
```

Comment:
No need to exists variable. Just use the function call in the if statement

Reply:
Done